engine/classes/Elgg/ActionsService.php

```diff
@@ -161,61 +180,80 @@ public function execute($action, $forwarder = "") {
 			return $forward('', ELGG_HTTP_OK);
 		}
 
-		$file = $this->actions[$action]['file'];
-
-		if (!is_file($file) || !is_readable($file)) {
-			ob_end_clean();
-			return $forward('actionnotfound', ELGG_HTTP_NOT_IMPLEMENTED);
-		}
-
 		// set the maximum execution time for actions
 		$action_timeout = $this->config->action_time_limit;
 		if (isset($action_timeout)) {
 			set_time_limit($action_timeout);
 		}
 
-		$result = Includer::includeFile($file);
+		$callback = $this->actions[$action]['callback'];
+		if ($callback) {
```

Comment:
If even filenames got wrapped in callback functions we wouldn't need this branch...

Reply:
True

Reply:
Naw. This results in far fewer objects created. It's fine.